build: `make sbsh-sb` fails on rerun with 'ln: failed to create hard link sb: File exists'

## Repro

Reproduced on `origin/main` at `d481c88`.

```bash
git clean -fdx -e .git
make sbsh-sb   # succeeds; produces sbsh + sb hardlink
make sbsh-sb   # fails on the ln step
```

Or, more practically, the same failure shows up any time `sbsh`/`sb` already exist in the repo root from an earlier build (e.g. you edited a Go file and rebuilt without running `make clean` in between).

## Observed

```
go build \
-o sbsh \
-ldflags="-s -w -X github.com/eminwux/sbsh/cmd/config.Version=v0.8.0-3-gd481c88-dirty" \
./cmd/
ln sbsh sb
ln: failed to create hard link 'sb': File exists
make: *** [Makefile:30: sbsh-sb] Error 1
```

`go build -o sbsh ./cmd/` overwrites `sbsh` happily, but the next line — `ln sbsh sb` — refuses to replace `sb`, because `ln` without `-f` errors on an existing destination.

## Root cause

`Makefile:28-33` (the canonical `sbsh-sb` target):

```makefile
sbsh-sb:
	go build \
	-o sbsh \
	-ldflags="-s -w -X $(MODULE)/cmd/config.Version=$(SBSH_VERSION)" \
	./cmd/
	ln sbsh sb
```

The `ln` invocation is non-idempotent. The project's contract (per the project root `CLAUDE.md`) is that `sbsh-sb` is *the* canonical local build target; CI is fine because it runs in a fresh checkout, but every contributor who runs the target twice in the same working tree without `make clean` hits this. Workaround today is `rm -f sbsh sb && make sbsh-sb` (or `make clean`, which also wipes `$HOME/.sbsh/run/*` — overkill).

## Suggested fix

Two-character change: `ln sbsh sb` → `ln -f sbsh sb`. `-f` is supported by both GNU coreutils `ln` and BSD `ln` (the project already targets `linux darwin freebsd` in its release matrix), so portability is a non-issue. Equivalent alternative: `rm -f sb && ln sbsh sb` — same effect, slightly noisier. Either makes the target idempotent.

--- a/Makefile
+++ b/Makefile
@@ -30,7 +30,7 @@ sbsh-sb:
 	-o sbsh \
 	-ldflags="-s -w -X $(MODULE)/cmd/config.Version=$(SBSH_VERSION)" \
 	./cmd/
-	ln sbsh sb
+	ln -f sbsh sb
 
 sbsh:
 	go build -o sbsh ./cmd/sbsh

